src/nodes/basic.ts

```diff
@@ -16,54 +30,106 @@ export const docComment: NodeSpec = {
   content: `(${NodeGroups.block} | ${NodeGroups.heading} | ${nodeNames.equation})+`, // browsers will completely collapse the node when it's empty `+` is necessary
 };
 
-export const paragraph: NodeSpec = {
+export const paragraph: MyNodeSpec<Attrs, Paragraph> = {
+  attrs: {},
   content: `${NodeGroups.inline}*`,
   group: NodeGroups.block,
   parseDOM: [{ tag: 'p' }],
   toDOM() {
     return ['p', 0];
   },
+  attrsFromMdastToken: () => ({}),
+  toMyst: (props): Paragraph => ({
+    type: 'paragraph',
+    children: (props.children || []) as PhrasingContent[],
+  }),
 };
 
-export const blockquote: NodeSpec = {
+export const blockquote: MyNodeSpec<Attrs, Blockquote> = {
+  attrs: {},
   content: `${NodeGroups.block}+`,
   group: NodeGroups.block,
   defining: true,
   parseDOM: [{ tag: 'blockquote' }],
   toDOM() {
     return ['blockquote', 0];
   },
+  attrsFromMdastToken: () => ({}),
+  toMyst: (props): Blockquote => ({
+    type: 'blockquote',
+    children: (props.children || []) as FlowContent[],
+  }),
 };
 
 /** Horizontal rule */
-export const horizontal_rule: NodeSpec = {
+export const horizontal_rule: MyNodeSpec<Attrs, ThematicBreak> = {
+  attrs: {},
   group: NodeGroups.block,
   parseDOM: [{ tag: 'hr' }],
   toDOM() {
     return ['hr', { class: 'break' }];
   },
+  attrsFromMdastToken: () => ({}),
+  toMyst: (): ThematicBreak => ({ type: 'thematicBreak' }),
 };
 
 export const text: NodeSpec = {
   group: NodeGroups.inline,
 };
 
-export const hard_break: NodeSpec = {
+export const hard_break: MyNodeSpec<Attrs, Break> = {
+  attrs: {},
   inline: true,
   group: NodeGroups.inline,
   selectable: false,
   parseDOM: [{ tag: 'br' }],
   toDOM() {
     return ['br'];
   },
+  attrsFromMdastToken: () => ({}),
+  toMyst: (): Break => ({ type: 'break' }),
+};
+
+export type ListAttrs = {
+  order: number | null;
 };
 
 const listNodes = addListNodes(
   OrderedMap.from({}),
   `paragraph ${NodeGroups.block}*`,
   NodeGroups.block,
-);
+) as OrderedMap<MyNodeSpec<any, any>>;
+
+export type OrderedListAttrs = {
+  order: number;
+};
+
+export const ordered_list = listNodes.get('ordered_list') as MyNodeSpec<OrderedListAttrs, List>;
+ordered_list.attrsFromMdastToken = (token: GenericNode) => ({ order: token.start || 1 });
```

Comment:
Can probably type this?

Reply:
I think it is start no?
https://executablebooks.github.io/myst-spec/myst.schema.html#list

Confused about the `order` prop.

Reply:
`order` is on the schema / prosemirror side - this is the number the list starts at: https://prosemirror.net/docs/ref/#schema-list.orderedList. It's `start` on the myst side.

I think maybe there is a problem a few lines down though...
```typescript
ordered_list.toMyst = 
  ...
  start: props.start || undefined,
  ...
});
```
should have `start: props.order || undefined` I think.